stackrox/stackrox
# ROX-17113: Add ManageStandardsModal to Compliance page without patch requests

## Description

1. Update integation test files:
    * Compliance.helpers.js adapt helper function from vulnmanagement
    * Compliance.selectors.js delete obsolete selectors
    * complianceDashboard.test.js call helper function

2. Rename and rewrite ComplianceDashboardPage.tsx file:
    * Merge header elements from Header.js file instead of adding more props.
    * Verify write access for Compliance resource.
    * Add initial GET /v1/compliance/standards request.
    * Add conditional rendering for `ManageStandardsModal` element (to make sure that Formik state is correctly reinitialized each time that it opens).

3. Rename and rewrite ComplianceDashboardTile.tsx file:
    * Remove obsolete support for CONTROLS entity type.
    * Replace global `URLService.getURL` call with container-scope entity path segments, to simplify clean up for future deletion of classic containers.
    * Replace `EntityTileLink` which sometimes has layout problems with classic button style.

4. Add ManageStandardsModal.tsx file:
    * Thanks to Van for suggesting pattern for Formik `onSubmit` function:
        https://github.com/stackrox/stackrox/blob/master/ui/apps/platform/src/Containers/VulnMgmt/List/Images/WatchedImagesDialog.tsx#L44-L78

5. Add ComplianceService.ts file:
    * `hideScanResults` property is optional until backend supports it
    * `patchComplianceStandard` request has status code 404 until backend supports it

### Residue

1. Investigate `style={{ '--min-tile-height': '160px' }}` prop commented out because TypeScript error.
2. Although spooky side effect to re-render links and widgets after scan does not affect REST request, it might also become the method to re-render after change to hideScanResults properties:
    https://github.com/stackrox/stackrox/blob/master/ui/apps/platform/src/Containers/Compliance/ScanButton.js#L75

## Checklist

## Testing Performed

### Integration testing

In ui folder: 

1. `export ROX_DISABLE_COMPLIANCE_STANDARDS=true`

2. `yarn start`

In ui/apps/platform folder:

1. `yarn cypress-open`

    * compliance/complianceDashboard.test.js
    * compliance/complianceList.test.js

### Manual testing

1. Visit /main/compliance

    * EntityTileLink layout problem on stagingdb
        ![EntityTileLink_stagingdb](https://github.com/stackrox/stackrox/assets/11862657/766f7589-d2d4-4471-a74e-11f00805bb89)

    * ComplianceDashboardTile: 0 entities scanned before click **Scan Environment**
        ![0_entities_scanned](https://github.com/stackrox/stackrox/assets/11862657/148a5c81-1bee-45de-98dd-9d28b2544d43)

    * ComplianceDashboardTile: non-zero entities scanned in dark theme
        ![entities_scanned_dark](https://github.com/stackrox/stackrox/assets/11862657/883c1d03-d664-4826-b7b2-7c600e58af8b)

    * ComplianceDashboardTile: non-zero entities scanned in light theme
        ![entities_scanned_light](https://github.com/stackrox/stackrox/assets/11862657/13833d42-e684-4b52-b306-bfe7fd86545b)

2. Click entity links to visit entity list pages

3. Click **Manage Standards** button

    * Click a check box: see **Save** button becomes enabled
        ![Modal](https://github.com/stackrox/stackrox/assets/11862657/f5a466e8-bb63-4d79-b1a4-c111bf11e300)

    * Click **Save**: see GET /v1/compliance/standards request only
        ![Save](https://github.com/stackrox/stackrox/assets/11862657/06ab15ad-5dd4-4a1e-9f4c-0de0ebc341f1)

    * Temporarily enable commented-out code, click 3 check boxes, and then click **Save**: see 404 status for PATCH /v1/compliance/standard request and warning alert.
        ![patch_404](https://github.com/stackrox/stackrox/assets/11862657/58b46515-318e-425a-86c3-de323c7d4f69)

    * Click **Cancel**: see no request (however, see TODO comment about possible edge case to investigate)
